### ISSUE TITLE
Breaking change introduced

### ISSUE THREAD

**Issue by person1123** [ramda/types#64]
Hi - looks like this commit broke these types: https://github.com/ramda/types/commit/3f8e31d884a1630f04eebb813f279c1bc3671c24

```
[!] (plugin typescript) RollupError: @rollup/plugin-typescript TS1139: Type parameter declaration expected.
../../../../node_modules/@types/ramda/node_modules/types-ramda/es/index.d.ts (4453:22)
4453 export function pick<const Names extends readonly [PropertyKey, ...PropertyKey[]]>(names: Names): <U extends Record<ElementOf<Names>, any>>(obj: U) => string extends keyof U ? Record<string, U[keyof U]> : Pick<U, ElementOf<Names>>;
```

**Comment by Harris-Miller** [comment]
This error is because you are on typescript < 5

`@types/ramda`, which is backed by `types-ramda` is labeled that [min support is typescript 5.0](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ramda/index.d.ts#L8)

Specifically, the [const Type Parameter](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-0.html#const-type-parameters) syntax is being used [here](https://github.com/ramda/types/blob/develop/types/pick.d.ts#L3)

**Comment by y-a-v-a** [comment]
Fixable by adding

```json
"resolutions": {
    "types-ramda": "0.29.4",
}
```

to `package.json`

**Comment by Harris-Miller** [comment]
@y-a-v-a if you do need to not consume the latest, that is a valid work-around for `yarn`. The `npm` equivalent is [overrides](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#overrides)

```json
"overrides": {
    "types-ramda": "0.29.4"
}
```

**Comment by stevenKirill** [comment]
Hello ! I have the same problem, my linter shows me error from this library, how can I disabled linter not to check this types ?

**Comment by Harris-Miller** [comment]
This MR should fix ts4 incompatibilities: https://github.com/ramda/types/pull/80

**Comment by whoisYeshua** [comment]
> This MR should fix ts4 incompatibilities: #80

We still have issue on `"typescript": "4.9.5"`. 

Looks like `typesVersions` should solve this problem (I saw it in your PR, but it didn't get into the result branch):
```json
  "types": "es/index.d.ts",
  "typesVersions": {
    "<5.0": { "*": ["ts4/es/index.d.ts"] },
    "*": { "*": ["es/index.d.ts"] }
  },
```

By now we use `overrides` method, but we live in an monorepo and only one package uses ramda (and in npm monorepo overrides only work in a root package - so its not best solution)
